### ISSUE TITLE
Cannot add data source console CLI after updating to CLI v2.1.0-beta.1 if sever is an older version

### ISSUE THREAD

**Issue by aperezleyva** [hasura/graphql-engine#7692]
When running the cli `v2.1.0-beta.1` against a server `v2.0.9`, and attempting to add a data source, the console does not connect the to the data source. Neither Terminal, Output, Problems, and Debug Console list any possible issues but Console does not display the data source being connected.

### Version Information
Server Version: `2.0.9`
CLI Version (for CLI related issue): `2.1.0-beta.1`

### Environment
Tested on locally hosted instance

### What is the expected behaviour?
Data source should be connected

### Keywords
cli, console, backwards compatibility, data source, not connecting

### What is the current behaviour?
Data source is not added 

### How to reproduce the issue?
1. Create a locally hosted project, with server version set to `v2.0.9` within the docker-compose.yml  and cli version at `v2.1.0-beta.1`

<details hidden>
<summary>Instructions with file examples</summary>
<br>

#### config.yml
```
version: 3
endpoint: http://localhost:8080
```
#### .env
```
HASURA_GRAPHQL_ADMIN_SECRET=secret
PG_USERNAME=postgres
PG_PASSWORD=postgres
```
#### docker-compose.yml
```
version: "3.8"
services:
  postgres:
    image: postgres:13
    ports:
      - 5432:5432
    restart: always
    volumes:
      - db_data:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: ${PG_PASSWORD}
      POSTGRES_USER: ${PG_USERNAME}
      POSTGRES_DB: postgres
  graphql-engine:
    image: hasura/graphql-engine:v2.0.9
    ports:
      - "8080:8080"
    depends_on:
      - "postgres"
      - "business-logic"
    restart: always
    environment:
      # documentation on hasura configuration environment variables can be found at https://hasura.io/docs/latest/graphql/core/deployment/graphql-engine-flags/reference.html#server-flag-reference
      HASURA_GRAPHQL_METADATA_DATABASE_URL: postgres://${PG_USERNAME}:${PG_PASSWORD}@postgres/postgres
      # enable the console served by server
      HASURA_GRAPHQL_ENABLE_CONSOLE: "false" # set to "false" to disable console
      # enable debugging mode. It is recommended to disable this in production
      HASURA_GRAPHQL_DEV_MODE: "true"
      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
      # uncomment next line to set an admin secret
      HASURA_GRAPHQL_ADMIN_SECRET: ${HASURA_GRAPHQL_ADMIN_SECRET}
      PG_DATABASE_URL: postgres://${PG_USERNAME}:${PG_PASSWORD}@postgres/postgres
volumes:
  db_data:
```
Within the terminal run
```
docker compose up -d
hasura console
```
</details>

### Screenshots or Screencast
<img width="931" alt="hasura console bug" src="https://user-images.githubusercontent.com/90622744/137605617-58680a30-8e3f-4d6c-8100-8057ce0138e3.png">

1. Navigate to the **Data & Schema Management tab**
2. Set a **Database Display Name**
3. Connect to database set to **Environment Variable** from the _docker-compose.yml_ file **PG_DATABASE_URL**
4. Console will appear as executing but will not connect to the data source

### Any possible solutions?
Rollback the CLI installation from `v2.1.0-beta.1` to `v2.0.9`
Possibly related to issue #7689

**Comment by rikinsk** [comment]
This was actually due to a bug in the console of versions `v2.0.9` and earlier as opposed to the CLI. This has now been fixed in `v2.1.0`.

**Comment by 0xMcsweeja** [comment]
happening for me still
